backends/graphics/surfacesdl/surfacesdl-graphics.cpp

```diff
@@ -1181,10 +1189,19 @@ void SurfaceSdlGraphicsManager::internUpdateScreen() {
 	}
 	if (_currentShakeYOffset != _gameScreenShakeYOffset ||
 		(_cursorNeedsRedraw && _mouseLastRect.y <= _currentShakeYOffset)) {
-		SDL_Rect blackrect = {0, 0, (Uint16)(_videoMode.screenWidth * _videoMode.scaleFactor), (Uint16)(_gameScreenShakeYOffset * _videoMode.scaleFactor)};
+		SDL_Rect blackrect = {0, 0, (Uint16)(_videoMode.screenHeight * _videoMode.scaleFactor), (Uint16)(_videoMode.screenHeight * _videoMode.scaleFactor)};
 
-		if (_videoMode.aspectRatioCorrection && !_overlayInGUI)
-			blackrect.h = real2Aspect(blackrect.h - 1) + 1;
+		if (_gameScreenShakeYOffset >= 0)
+			blackrect.h = (Uint16)(_gameScreenShakeYOffset * _videoMode.scaleFactor);
+		else {
+			blackrect.h -= ((-_gameScreenShakeYOffset) * _videoMode.scaleFactor);
+			blackrect.y = ((-_gameScreenShakeYOffset) * _videoMode.scaleFactor);
```

Comment:
And here again I have some doubts on the logic for the black rect size and position in the case of a negative offset. To me you seem to compute the non-black rect instead.

Reply:
Done